cmd/internal/search_batch.go

```diff
@@ -28,6 +34,73 @@ var (
 	flagThreshold    = flag.Float64("threshold", 0.99, "Minimum match percentage required for blocking")
 )
 
+func SearchBatch(logger log.Logger) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		err := r.ParseMultipartForm(128 << 20) // 128 MB limit for file size
+		if err != nil {
+			http.Error(w, "Unable to parse form", http.StatusBadRequest)
+			return
+		}
+		setFlagsFromFormFields(r)
+
+		file, handler, err := r.FormFile("csvFile")
+		if err != nil {
+			http.Error(w, "Unable to get file", http.StatusBadRequest)
+			return
+		}
+		defer file.Close()
+
+		input, err := io.ReadAll(file)
+		if err != nil {
+			http.Error(w, "Unable to read file content", http.StatusInternalServerError)
+			return
+		}
+
+		rows := strings.Split(string(input), "\n")
+		conf := Config(DefaultApiAddress, true)
+		api := moov.NewAPIClient(conf)
+		result, err := ProcessRows(rows, api, logger)
+		if err != nil {
+			http.Error(w, "Unable to process input", http.StatusInternalServerError)
+			return
+		}
+		output := strings.Join(result, "\n")
+
+		w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", handler.Filename))
+		w.Header().Set("Content-Type", "text/csv")
+		w.Header().Set("Content-Length", fmt.Sprint(len(output)))
+
+		_, err = w.Write([]byte(output))
+		if err != nil {
+			http.Error(w, "Unable to write response", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+func setFlagsFromFormFields(r *http.Request) {
+	if threshold := r.FormValue("threshold"); threshold != "" {
+		if f, err := strconv.ParseFloat(threshold, 64); err == nil {
+			*flagThreshold = f
```

Comment:
What is this doing? Is `flagThreshold` scoped to the request?

Reply:
Good question! I should probably find a more appropriate way to represent the parameters here; in the interest of speed I just moved the command line parameters over.

Running the Watchman server locally, I ran into the problem that the values persist between calls. (e.g. send threshold=0.6 in first call, then don't send in second call = it uses 0.6)

I did this hacky approach for now. If the POST request doesn't include a parameter, set `flagX` to the default value [to immitate the CLI behavior].

I will definitely do this differently in our own OFAC code.